test/classification/training/test_trainer.py

```diff
@@ -90,6 +91,94 @@ def test_trainer_errors(self):
         with self.assertRaises(ValueError):
             trainer.train_model(model, [dataloader])
 
+    def test_checkpointer_init(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            trainer = Trainer(
+                **trainer_config, checkpointing=True, checkpoint_dir=temp_dir
+            )
+            trainer.train_model(model, [dataloaders[0]])
+            self.assertIsNotNone(trainer.checkpointer)
+
+            trainer = Trainer(
+                **trainer_config,
+                checkpointing=True,
+                logging=True,
+                log_dir=temp_dir,
+                checkpoint_dir=None,
+            )
+            trainer.train_model(model, [dataloaders[0]])
+            self.assertIsNotNone(trainer.checkpointer)
+            self.assertEqual(
+                trainer.checkpointer.checkpoint_dir, trainer.log_writer.log_dir
+            )
+
+    def test_log_writer_init(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            trainer = Trainer(
+                **trainer_config, logging=True, log_dir=temp_dir, writer="json"
+            )
+            trainer.train_model(model, [dataloaders[0]])
+            self.assertIsInstance(trainer.log_writer, LogWriter)
+
+            trainer = Trainer(
+                **trainer_config, logging=True, log_dir=temp_dir, writer="tensorboard"
+            )
+            trainer.train_model(model, [dataloaders[0]])
+            self.assertIsInstance(trainer.log_writer, TensorBoardWriter)
+
+            with self.assertRaises(ValueError):
+                trainer = Trainer(
+                    **trainer_config, logging=True, log_dir=temp_dir, writer="foo"
+                )
+                trainer.train_model(model, [dataloaders[0]])
+
+    def test_optimizer_init(self):
+        trainer = Trainer(**trainer_config, optimizer="sgd")
+        trainer.train_model(model, [dataloaders[0]])
+        self.assertIsInstance(trainer.optimizer, optim.SGD)
+
+        trainer = Trainer(**trainer_config, optimizer="adam")
+        trainer.train_model(model, [dataloaders[0]])
+        self.assertIsInstance(trainer.optimizer, optim.Adam)
+
+        trainer = Trainer(**trainer_config, optimizer="adamax")
+        trainer.train_model(model, [dataloaders[0]])
+        self.assertIsInstance(trainer.optimizer, optim.Adamax)
+
+        with self.assertRaises(ValueError):
```

Comment:
For the `ValueErrors`, can we include checking the error message as a check for consistency
```python
with self.assertRaisesRegex(ValueError, "not currently implemented"):
```